Edge BYOOS (#1277)
spectrocloud/librarium: content/docs/04-clusters/03-edge/09-site-deployment/01-model-profile.md

Reviewer: @rahulhazra97 I'm a little unclear in this doc if we want to call it the more general BYOOS pack or be more specific in this Edge doc and call it BYOS Edge OS.

I understand there's an Edge and non-Edge version of BYOOS - I'm just not sure how we plan to present them in the docs yet.

4. Click on **Add Cluster Profile**. 

5. Provide **Basic Information**, such as profile name, description, and tags. Select **Full** and click on **Next**.

6. Select **Edge Native** as the **Cloud Type** and click on **Next**.

7. Select **Public Repo** in the **Registry field**.

8. Select **BYOS Edge OS** in the **Pack Name** field and the pack version. 

9. Click on the code editor button  **</\>** to open up the editor

  <br />

  ![A view of the Kubernetes pack editior with a YAML configuration](/clusters_site-deployment_model-profile_byoos-pack-yaml.png)

10. In the pack editor, you must update the `system.uri` parameter. Use the custom OS image you created in the EdgeForge process. Refer to the EdgeForge [Build Images](/clusters/edge/edgeforge-workflow/build-images) guide if you are missing a custom OS image. The following is an example configuration using a custom OS image.

  <br />

  ```yaml
  pack:
  content:
    images: 
      - image: '{{.spectro.pack.edge-native-byoi.options.system.uri}}'
      # - image: example.io/my-other-images/example:v1.0.0 
      # - image: example.io/my-super-other-images/example:v1.0.0 
      
  options: 

>     system.uri: example.io/my-images/example-custom-os:v1.4.5

Author: I will verify this Rita.

>     system.uri: example.io/my-images/example-custom-os:v1.4.5

Reviewer 2: It's an example.

Optionally, add additional Helm or OCI registries and include applications hosted in those registries in add-on profiles. Check out the guide for adding a [Helm](/registries-and-packs/helm-charts) or [OCI](/registries-and-packs/oci-registry) registry to learn more.

> # Build Custom Cluster Profile With BYOOS Pack

Reviewer: ```suggestion
# Create a Custom Cluster Profile With BYOOS Pack
```

# Prerequisites

- Provider images are built.

- ` system.uri`should be constrcuted. 

# Enablement

1. Log in to [Palette](https://console.spectrocloud.com) as a **Tenant Admin**.

2. Navigate to the left **Main Menu** and select **Profiles**.

3. Click on **Add Cluster Profile**. 

4. Provide **Basic Information**, such as profile name and description. Select **Full** and click on **Next**. <br /> <br />

    <InfoBox>

    Refer to the [Creating Cluster Profiles](/https://docs.spectrocloud.com/cluster-profiles/task-define-profile) resource to learn about creating a new cluster profile.

    </InfoBox>

5. Select **Edge Native** as the **Cloud Type** and click on **Next**.

6. Select **Public Repo** in the **Registry field**.

7. Select **BYOOS Edge OS** in the **Pack Name** field and the pack version. 

8. The pack editor displays on the right section of the dashboard where you can customise the BYOOS pack. 

## Installer Parameters

| Parameter            | Description                                            |
|----------------------|--------------------------------------------------------|
| `pack:content:` | Specifies the content of the BYOOS pack. |

> | `pack.content.images` | Specifies the list of images to use for the OS pack. |

Reviewer: ```suggestion
| `pack.content.images` | Specifies the list of OS images to use for the pack. |
```

> | `pack.content.images.-  images` | Specifies an image to use for the OS pack. |

Reviewer: In lines 141 and 142, my understanding is that the first parameter provides a list of OS images that can be used, and the second specifies just one image from the list to use. Is that correct?

> | `pack.content.images.-  images` | Specifies an image to use for the OS pack. |

Author: Yes, that's right. Thanks for your suggestion, Rita.

## User Data Parameters

| Parameter            | Description                                            |
|----------------------|--------------------------------------------------------|
| `options.system.uri:` | The system URI specifying the location of the BYOOS image. |
| `image.registry` | The domain of the registry where the BYOOS image is stored. |
| `image.repo` | The name of the BYOOS image repository. |
| `image.palette.edge.version` | The Palette Edge software version used in the BYOOS container image. |
| `image.client.tag` |  The tag given to the image used for the BYOOS, specifying its version. |

Example BYOOS Pack Customization File

<br />

```yaml
pack:
 content:
   images: 
    - image: ‘{{.spectro.pack.edge-native-byoi.options.system.uri}}’
    
options: 
 system.uri: “{{.spectro.pack.edge-native-byoi.image.registry}}/{{.spectro.pack.edge-native-byoi.image.repo}}:{{.spectro.system.kunernetes.version}}-{{.spectro.pack.edge-native-byoi.image.version}}_{{.spectro.pack.edge-native-byoi.image.client.tag}}”

image: 
 registry: "" 
 repo: ""
 palette.edge.version: ""
 client.tag: ""
```

<br />

> Refer to the  [BYOOS Pack](/) resource to learn more about the pack details.

Reviewer: Missing doc URL. 
Also, how are we referring to the pack? Are we just calling it the BYOOS pack because there's one for Edge and one for non-Edge?

> Refer to the  [BYOOS Pack](/) resource to learn more about the pack details.

Reviewer: ```suggestion
Refer to the [BYOOS Pack](/) resource to learn more about the pack details.
```

<br />

> 9. Review your own custom BYOOS pack, and close the updated pack editor. Click on the **Next layer** to configure the Kubernetes layer. 

Reviewer: ```suggestion
9. Review your custom BYOOS pack, and close the updated pack editor. Click on the **Next layer** to configure the Kubernetes layer. 
```

You have successfully configured the BYOOS pack to update your custom images to build the Edge cluster profile. 

# Validation

> Verify you created a cluster profile for Edge hosts by using the following steps.

Reviewer: Let's add a break after this to add space between the intro sentence and steps. 
Steps in the Validation should be double-spaced.

Verify you created a cluster profile for Edge hosts by using the following steps.

1. Log in to [Palette](https://console.spectrocloud.com).

2. Choose the desired scope, project or **Tenant Admin**.

3. Navigate to the left **Main Menu** and select **Profiles**.

4. Use the **Cloud Types drop-down Menu** and select **Edge Native**.

> 5. Your newly created cluster profile is displayed along with other cluster profiles of the same type.

Reviewer: Step 5 is more of a result, not a step.

> 5. Your newly created cluster profile is displayed along with other cluster profiles of the same type.

Reviewer: ```suggestion
You can view your newly created cluster profile on the **Cluster Profiles** page.
```
